tests/flakes.sh

```diff
@@ -274,3 +281,51 @@ nix build -o $TEST_ROOT/result --flake-registry $registry flake3:sth
 mv $flake2Dir.tmp $flake2Dir
 mv $nonFlakeDir.tmp $nonFlakeDir
 nix build -o $TEST_ROOT/result --flake-registry $registry flake3:xyzzy flake3:fnord
+
+# Test doing multiple `lookupFlake`s
+nix build -o $TEST_ROOT/result --flake-registry $registry flake4:xyzzy
+nix build -o $TEST_ROOT/result --flake-registry $registry file://$flake4Dir:xyzzy
+
+# Make branch "removeXyzzy" where flake3 doesn't have xyzzy anymore
+git -C $flake3Dir checkout -b removeXyzzy
+rm $flake3Dir/flake.nix
+
+cat > $flake3Dir/flake.nix <<EOF
+{
+  name = "flake3";
+
+  epoch = 201906;
+
+  inputs = [ "flake1" "flake2" ];
+
+  nonFlakeInputs = {
+    nonFlake = "$nonFlakeDir";
+  };
+
+  description = "Fnord";
+
+  outputs = inputs: rec {
+    packages.sth = inputs.flake1.packages.foo;
+    packages.fnord =
+      with import ./config.nix;
+      mkDerivation {
+        inherit system;
+        name = "fnord";
+        buildCommand = ''
+          cat \${inputs.nonFlake}/README.md > \$out
+        '';
+      };
+  };
+}
+EOF
+git -C $flake3Dir add flake.nix
+git -C $flake3Dir commit -m 'Remove packages.xyzzy'
+
+# Check that building flake4 still works
+nix build -o $TEST_ROOT/result --flake-registry $registry file://$flake4Dir:xyzzy
+
+# Test whether fuzzy-matching works for IsAlias
+(! nix build -o $TEST_ROOT/result --flake-registry $registry flake4/removeXyzzy:xyzzy)
+
+# Test whether fuzzy-matching works for IsGit
+(! nix build -o $TEST_ROOT/result --flake-registry $registry file://$flake4Dir/removeXyzzy:xyzzy)
```

Comment:
It's probably a good idea to also have a *positive* test for `flake4/removeXyzzy`, e.g. `nix build flake4/removeXyzzy:sth`.